### PR TITLE
Adjust problems describing property validation

### PR DESCRIPTION
**Pull request by donat**
The contextual message in problem reports were misaligned in a lot of cases as the displayName overrode the value.
I've added `TODO`s to all code that emits a property validation problem report without corresponding test coverage.

### CODE REVIEW

**Comment by donat** [comment]
@bot-gradle test QuickFeedbackLinux

**Comment by donat** [comment]
@bot-gradle test QuickFeedbackLinux

**Comment by donat** [comment]
@bot-gradle test RFM

**Comment by reinsch82** [comment]
@bot-gradle test this

File: platforms/core-configuration/model-core/src/main/java/org/gradle/internal/reflect/validation/DefaultTypeAwareProblemBuilder.java [hunk 1]
```diff
@@ -67,6 +73,67 @@ public TypeAwareProblemBuilder parentProperty(@Nullable String parentProperty) {
...
+    public static String introductionFor(Map<String, Object> additionalMetadata) {
+        StringBuilder builder = new StringBuilder();
+        String rootType = ofNullable(additionalMetadata.get(TYPE_NAME))
+            .map(Object::toString)
+            .filter(DefaultTypeAwareProblemBuilder::shouldRenderType)
+            .orElse(null);
+        DefaultPluginId pluginId = ofNullable(additionalMetadata.get(PLUGIN_ID))
+            .map(Object::toString)
+            .map(DefaultPluginId::new)
+            .orElse(null);
```
Comment (reinsch82) [inline platforms/core-configuration/model-core/src/main/java/org/gradle/internal/reflect/validation/DefaultTypeAwareProblemBuilder.java:93]: :thinking: we should not use the `orElse` here and instead keep the `Optional` and use `isPresent` down the line.

**Review by reinsch82 — APPROVED** [review]
LGTM
will add coverage later

**Review by hegyibalint — APPROVED** [review]

**Pull request merged**